src/components/inputs/radio-input/radio-option.js

```diff
@@ -1,74 +1,82 @@
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
-    // This prop forces Radio.Option to be rendered in a hovered state (thought isDisabled takes
-    // precedence over that). We need that to address a use-case when hovering is comming
-    // from somewhere up the hierarchy. There is no need to touch this prop in case
-    // all you need is a general highlighting on hover of Radio.Option body, which is solved
-    // by a corresponding :hover selector in the syles of this component.
-    isHovered: PropTypes.bool,
-    isDisabled: PropTypes.bool,
-    isChecked: PropTypes.bool,
-    children: PropTypes.node,
-    // Injected through as compound component
-    // not required as `createElement` is used.
-    name: PropTypes.string,
-    onChange: PropTypes.func,
-  };
-  static defaultProps = {
-    isDisabled: false,
-  };
-
-  render() {
-    return (
-      <div>
-        <label css={getLabelStyles(this.props)}>
-          <Spacings.Inline alignItems="center">
-            <div css={getContainerStyles(this.props)}>
-              {this.props.isChecked ? <Icons.Checked /> : <Icons.Default />}
-            </div>
-            {this.props.children && (
-              <div>
-                <Text.Body
-                  // FIXME: add proper tones when we have disabled/primary in tones
-                  tone={this.props.isDisabled ? 'secondary' : undefined}
-                  // We need isInline to prevent Text.Body from rendering a p tag.
-                  // Otherwise we would get a validateDomNesting warning because
-                  // the "children" might contain a div (e.g for a tooltip), so we
-                  // would wrongly render a div inside a p tag.
-                  isInline={true}
-                >
-                  {this.props.children}
-                </Text.Body>
-              </div>
-            )}
-            <input
-              css={css`
-                display: none;
-              `}
-              name={this.props.name}
-              value={this.props.value}
-              onChange={this.props.onChange}
-              disabled={this.props.isDisabled}
-              checked={this.props.isChecked}
-              type="radio"
-              {...filterDataAttributes(this.props)}
-            />
-          </Spacings.Inline>
-        </label>
+const Option = props => (
+  <label
+    css={getLabelStyles(props)}
+    role="radio"
+    aria-checked={props.isChecked}
+    onFocus={props.onFocus}
+    onBlur={props.onBlur}
+    htmlFor={props.id}
+  >
+    <Spacings.Inline scale="s" alignItems="center">
+      <div css={getContainerStyles(props)}>
+        {props.isChecked ? <Icons.Checked /> : <Icons.Default />}
+      </div>
+      <div
+        css={css`
+          width: 100%;
+          font-size: 1rem;
+          font-family: ${vars.fontFamilyDefault};
+          color: ${props.isDisabled
+            ? vars.fontColorDisabled
+            : vars.fontColorDefault};
+        `}
```

Comment:
So I ended up removing the `Text.Body` wrapper, as it was just causing problems with the HTML markup (e.g. `<div>` nested inside a `<p>`).
The styles to apply are actually just a few, I think it's better this way.

/cc @montezume